python/plugins/processing/modeler/ModelerDialog.py

```diff
@@ -242,9 +241,11 @@ def openModel(self):
             )
             self.loadModel(filename)
 
-    def repaintModel(self, showControls=True):
+    def repaintModel(self, showControls=True, initialPaint=False):
+        _oldViewportPolygon = self.view().mapToScene(self.view().viewport().rect())
+        oldViewportRect = _oldViewportPolygon.boundingRect()
```

Comment:
```suggestion
        old_viewport_rect = self.view().mapToScene(self.view().viewport().rect()).boundingRect()
```

But do we actually need the call to boundingRect() here? QGraphicsView.mapToScene should return a QRect when called with a QRect argument already...

Reply:
Unfortunalty mapToScene return a `QPolygonF` so the `.boundingRect()` just convert it back a to a `QRect`

Reply:
Actually I simplified the code to avoid the need to check for `initialPaint` and I could also achieve the same result with just a point with `centerOn` instead of `fitInView`

see 55f2cbe